R/utils.R

```diff
@@ -259,84 +259,333 @@ rowcol_to_dim <- function(row, col) {
+    if (any(is_cols_a_colname)) {
+      if (length(is_cols_a_colname) != 1) {
+        stop(
+          "Supplying multiple column names is not supported by the `wb_dims()` helper, use the `cols`  arguments instead.",
+          "\n Use a single `cols` at a time with `wb_dims()`"
+        )
+      }
+      col_name <- cols_arg
+
+      cols_arg <- which(colnames(x) == cols_arg)
+      message("Transforming col name = '", col_name, "' to `cols = ", cols_arg, "`")
+    }
+  }
 
+  if (!is.null(rows_arg)) {
+    assert_class(rows_arg, class = "integer", envir = parent.frame(n = 2), arg_nm = "rows_arg")
   }
 
-  rows <- rows + srow
-  cols <- cols + scol
+  if (!is.null(cols_arg)) {
+    cols_arg <- col2int(cols_arg)
+    assert_class(cols_arg, class = "integer", envir = parent.frame(n = 2), arg_nm = "cols_arg")
+  }
+
+  srow <- args$start_row %||% 1L
+  srow <- srow - 1L
+  scol <- col2int(args$start_col, allow_null = TRUE) %||% 1L
+  scol <- scol - 1L
+  # after this point, no assertion, assuming all elements to be acceptable
+
+
 
-  if (length(rows) == 1 && length(cols) == 1) {
+  col_names <- args$col_names
+  row_names <- args$row_names
+
+  rows_arg
+  cols_arg
+
+  if (!all(length(scol) == 1, length(srow) == 1, scol >= 0, srow >= 0)) {
+    stop("Internal error. At this point scol and srow should have length 1.")
+  }
+
+
+  # if `!x` return early
+  if (!x_present) {
+    row_span <- srow + rows_arg %||% 1L
+    col_span <- scol + cols_arg %||% 1L
+    if (length(row_span) == 1 && length(col_span) == 1) {
+      # A1
+      row_start <- row_span
+      col_start <- col_span
+      dims <- rowcol_to_dim(row_start, col_start)
+    } else {
+      # A1:B2
+
+      dims <- rowcol_to_dims(row_span, col_span)
+    }
+    return(dims)
+  }
+  # Making sure that at this point, we only cover the case for `x`
+
+  col_names <- col_names %||% x_has_named_dims
+  row_names <- row_names %||% FALSE
+  if (!col_names && row_names && x_has_named_dims) {
+    warning("The combination of `row_names = TRUE` and `col_names = FALSE` is not recommended.",
+      "`col_names` allows to select the region that contains the data only.",
+      "`row_names` = TRUE adds row numbers if the data doesn't have rownames.",
+      call. = FALSE
+    )
+  }
+  assert_class(col_names, "logical")
+  assert_class(row_names, "logical")
+  x <- as.data.frame(x)
+
+  nrow_to_span <- nrow(x)
+  ncol_to_span <- ncol(x)
+
+  if (col_names && x_has_named_dims) {
+    nrow_to_span <- nrow_to_span + 1
+  }
+  # if without column names and with named dimensions
+  # We will increment the start row by 1.
+  if (!col_names && x_has_named_dims) {
+    srow <- srow + 1
+  }
+
+  # Adding a column when spanning.
+  if (row_names) {
+    ncol_to_span <- ncol_to_span + 1
+  }
+
+  # if (row_names) {
+  #   scol <- scol + 1
+  # }
+
+  if (!all(scol >= 0, srow >= 0)) {
+    stop("Internal error. At this point `start_col` and `start_row` should have length 1.")
+  }
```

Comment:
Haha, you really like to catch stuff 👍

Reply:
I often add these conditions for myself  because I want to be bad input later. I want to assume everything is going well after a certain point to avoid potential messy failures.